### ISSUE TITLE
Calling LocalizationProvider.Translate<T>() returns null if none of the translations exist for language requested

### ISSUE THREAD

**Issue by patkleef** [valdisiljuconoks/LocalizationProvider#154]
Calling the Translate (for FR) method returns null when all properties of the class are not entered in the UI. I would expect it fallbacks to the default language. If one of the properties are entered it uses the correct behavior, all properties that are not entered for the FR language are fallback to the EN language.

`_provider.Translate<PaginationComponentResources>()`

```
[LocalizedResource]
    public class PaginationComponentResources
    {
        public string SearchButton { get; set; } = "Search";
        [TranslationForCulture("Previous Page", "en")]
        public string Previous { get; set; } = "Previous Page";
        [TranslationForCulture("Next Page", "en")]
        public string Next { get; set; } = "Next Page";
    }
```

**Comment by valdisiljuconoks** [comment]
just to clarify, you are requesting translations for `fr`, none of the properties are translated in language requested -> return `null`. correct?

and when at least one of the properties is translated, object is returned with fallback to `en` for the rest of the not translated properties?

**Comment by patkleef** [comment]
Yes, that is correct

**Comment by valdisiljuconoks** [comment]
`ConfigurationContext.EnableInvariantCultureFallback` is `false` right?
it's quite tricky now with default Asp.Net Mvc (has no sense of fallback) + Episerver (has its own fallback schema) configuration settings to balance between these 2.
I'll see what I can imagine here..

**Comment by patkleef** [comment]
I don't set the EnableInvariantCultureFallback explicitly so I suppose it's the default value.

**Comment by valdisiljuconoks** [comment]
smaller fix will be out as part of 5.6. proper fallback respect is coming in 5.7 (it will be bigger release affecting all supported platforms).
